File: for-local-test/387.first-unique-character-in-a-string.java
class Solution {
    public int firstUniqChar(String s) {
        int occurrences[] = new int[26];
        for (char c : s.toCharArray()){
            occurrences[c - 'a']++;
        }
        for (char c : s.toCharArray()){
            if(occurrences[c - 'a'] == 1) return s.indexOf(c);
        }
        return -1;
    }
    public static void main(String[] args) {
        Solution solution = new Solution();
        String s = "leetcode";
        int result = solution.firstUniqChar(s);
        System.out.println("The index of the first unique character in \"" + s + "\" is: " + result);
    }
}

File: for-local-test/python/387.first-unique-character-in-a-string.py
class Solution:
    def firstUniqChar(self, s: str) -> int:
        occurences = [0] * 26
        for c in s:
            occurences[ord(c) - ord('a')] += 1
        for c in s:
            if occurences[ord(c) - ord('a')] == 1:
                return s.index(c)
        return -1

if __name__ == "__main__":
    solution = Solution()
    s = "leetcode"
    result = solution.firstUniqChar(s)
    print(f"The index of the first unique character in \"{s}\" is: {result}")